core/src/main/java/com/segment/analytics/kotlin/core/Telemetry.kt

```diff
@@ -76,7 +77,14 @@ object Telemetry: Subscriber {
     var host: String = Constants.DEFAULT_API_HOST
     // 1.0 is 100%, will get set by Segment setting before start()
     // Values are adjusted by the sampleRate on send
-    var sampleRate: Double = 1.0
+    @Volatile private var _sampleRate: Double = 1.0
+    var sampleRate: Double
+        get() = _sampleRate
+        set(value) {
+            synchronized(this) {
```

Comment:
we should try our best to avoid using `synchronized` since our API is coroutine-based. people could use our API inside a coroutine which causes a suspend function to block.